From 10f5e44604e163383f8f32ab05307edd69ecab1c Mon Sep 17 00:00:00 2001
From: Hartmut Goebel <h.goebel@crazy-compilers.com>
Date: Fri, 3 Mar 2017 16:56:47 +0100
Subject: [PATCH] Fix logging: Don't defer initialization of logger.

Deferring the initialization led to some curious problems when running
py.test with plugin pytest-catchlog: PyIModuleGraph's TRACE logs
where logged, even if log-level is DEBUG.

This was caused by logger "PyInstaller.depend.analysis" being set up
prior to logger "PyInstaller" which made the former's parent to be the
 root-logger. But ``--log-level=DEBUG`` only effects the "PyInstaller"
logger, not the root-logger, and thus "PyInstaller.depend.analysis"'
level as unset.

And logger "PyInstaller.depend.analysis" was set up first, since we call
initialize_modgraph() as a session fixture.
---
 PyInstaller/__main__.py                      | 1 -
 PyInstaller/log.py                           | 9 ++-------
 PyInstaller/utils/cliutils/archive_viewer.py | 2 --
 PyInstaller/utils/cliutils/bindepend.py      | 2 --
 PyInstaller/utils/cliutils/grab_version.py   | 2 --
 PyInstaller/utils/cliutils/makespec.py       | 2 --
 PyInstaller/utils/cliutils/set_version.py    | 2 --
 7 files changed, 2 insertions(+), 18 deletions(-)

diff --git a/PyInstaller/__main__.py b/PyInstaller/__main__.py
index 1e9cd43..a4a47bd 100644
--- a/PyInstaller/__main__.py
+++ b/PyInstaller/__main__.py
@@ -56,7 +56,6 @@ def run(pyi_args=None, pyi_config=None):
     pyi_args     allows running PyInstaller programatically without a subprocess
     pyi_config   allows checking configuration once when running multiple tests
     """
-    PyInstaller.log.init()
     check_requirements()
 
     try:
diff --git a/PyInstaller/log.py b/PyInstaller/log.py
index 5fd0799..93b81ff 100644
--- a/PyInstaller/log.py
+++ b/PyInstaller/log.py
@@ -21,13 +21,8 @@ TRACE = logging.TRACE = DEBUG - 5
 logging.addLevelName(TRACE, 'TRACE')
 
 FORMAT = '%(relativeCreated)d %(levelname)s: %(message)s'
-
-
-def init():
-    # Allow deferring initialization
-    global logger
-    logging.basicConfig(format=FORMAT, level=logging.INFO)
-    logger = getLogger('PyInstaller')
+logging.basicConfig(format=FORMAT, level=logging.INFO)
+logger = getLogger('PyInstaller')
 
 
 def __add_options(parser):
diff --git a/PyInstaller/utils/cliutils/archive_viewer.py b/PyInstaller/utils/cliutils/archive_viewer.py
index 5cf72a5..28c6bf5 100644
--- a/PyInstaller/utils/cliutils/archive_viewer.py
+++ b/PyInstaller/utils/cliutils/archive_viewer.py
@@ -232,8 +232,6 @@ class ZlibArchive(pyimod02_archive.ZlibArchiveReader):
 
 
 def run():
-    PyInstaller.log.init()
-
     parser = argparse.ArgumentParser()
     parser.add_argument('-l', '--log',
                         default=False,
diff --git a/PyInstaller/utils/cliutils/bindepend.py b/PyInstaller/utils/cliutils/bindepend.py
index ac9b0c2..6890f26 100644
--- a/PyInstaller/utils/cliutils/bindepend.py
+++ b/PyInstaller/utils/cliutils/bindepend.py
@@ -24,8 +24,6 @@ import PyInstaller.log
 
 
 def run():
-    PyInstaller.log.init()
-
     parser = argparse.ArgumentParser()
     PyInstaller.log.__add_options(parser)
     parser.add_argument('filenames', nargs='+',
diff --git a/PyInstaller/utils/cliutils/grab_version.py b/PyInstaller/utils/cliutils/grab_version.py
index dddb937..646a8ff 100644
--- a/PyInstaller/utils/cliutils/grab_version.py
+++ b/PyInstaller/utils/cliutils/grab_version.py
@@ -16,8 +16,6 @@ import PyInstaller.log
 
 
 def run():
-    PyInstaller.log.init()
-
     parser = argparse.ArgumentParser(
         epilog = ('The printed output may be saved to a file, edited and '
                   'used as the input for a version resource on any of the '
diff --git a/PyInstaller/utils/cliutils/makespec.py b/PyInstaller/utils/cliutils/makespec.py
index 1e03879..908e941 100644
--- a/PyInstaller/utils/cliutils/makespec.py
+++ b/PyInstaller/utils/cliutils/makespec.py
@@ -21,8 +21,6 @@ import PyInstaller.log
 
 
 def run():
-    PyInstaller.log.init()
-
     p = argparse.ArgumentParser()
     PyInstaller.building.makespec.__add_options(p)
     PyInstaller.log.__add_options(p)
diff --git a/PyInstaller/utils/cliutils/set_version.py b/PyInstaller/utils/cliutils/set_version.py
index 72b5fd9..7d79559 100644
--- a/PyInstaller/utils/cliutils/set_version.py
+++ b/PyInstaller/utils/cliutils/set_version.py
@@ -14,8 +14,6 @@ import argparse
 import PyInstaller.log
 
 def run():
-    PyInstaller.log.init()
-
     parser = argparse.ArgumentParser()
     parser.add_argument('info_file', metavar='info-file',
                         help="text file containing version info")
-- 
2.10.1.windows.1

